lightning: add usage restriction for tidb-lightning (#7562)
pingcap/docs: tidb-lightning/tidb-lightning-distributed-import.md

Because TiDB Lightning needs to upload the generated Key-Value data to the TiKV node where each copy of the corresponding Region is located, the import speed is limited by the size of the target cluster. It is recommended to ensure that the number of TiKV instances in the target TiDB cluster and the number of TiDB Lightning instances are greater than n:1 (n is the number of copies of the Region). At the same time, you need to meet the following requirements to achieve the optimal import performance:

> - Deploy each TiDB Lightning instance on a dedicated machine. TiDB Lightning consumes all CPU resources by default. Deploying multiple instances on a single machine can not improve performance.

Reviewer: ```suggestion
- Deploy each TiDB Lightning instance on a dedicated machine. Since one TiDB Lightning instance consumes all CPU resources by default, deploying multiple instances on a single machine cannot improve performance.
```

- Example 1: Use Dumpling + TiDB Lightning to import sharded databases and tables into TiDB in parallel
- Example 2: Import single tables in parallel

> ### Restrictions

Reviewer: 上面第 46 行前还要加个限制条件：https://github.com/pingcap/docs-cn/pull/8249/files#diff-22740891a93250654c0a5076495140651bc61998ea472e52ff8e06419a8e0300R46

Author: 好的 加上了

TiDB Lightning exclusively uses some resources when running. If you need to deploy multiple TiDB Lightning instances on a single machine (which is not recommended for production environments), or on a disk shared by multiple machines, you need to be aware of the following usage restrictions.

> - You must set the `tikv-importer.sorted-kv-dir` to a different path for each TiDB Lightning instance. Multiple instances sharing the same path can lead to unintended behavior and may result in import failures or data errors.

Reviewer: ```suggestion
- Set `tikv-importer.sorted-kv-dir` to a unique path for each TiDB Lightning instance. Multiple instances sharing the same path can lead to unintended behavior and may result in import failures or data errors.
```

- Store each TiDB Lightning checkpoint separately. For more information about checkpoint configurations, see [TiDB Lightning Checkpoints](/tidb-lightning/tidb-lightning-checkpoints.md).
    - If you set checkpoint.driver = "file" (default), make sure that the path to the checkpoint is unique for each instance.

>     - If you set checkpoint.driver = "mysql", you need to set a different schema for each instance.

Reviewer: ```suggestion
    - If you set checkpoint.driver = "mysql", you need to set a unique schema for each instance.
```

> - The log file for each TiDB Lightning should be set to a different path. Sharing the same log file will impact log querying and troubleshooting.

Reviewer: ```suggestion
- The log file for each TiDB Lightning should be set to a unique path. Sharing the same log file will impact log querying and troubleshooting.
```

> - If you use the [Web Interface](/tidb-lightning/tidb-lightning-web-interface.md) or Debug API, you need to set `lightning.status-addr` to a unique address for each instance; otherwise the TiDB Lightning process can not start due to port conflict.

Reviewer: ```suggestion
- If you use the [Web Interface](/tidb-lightning/tidb-lightning-web-interface.md) or Debug API, you need to set `lightning.status-addr` to a unique address for each instance; otherwise, the TiDB Lightning process fails to start due to port conflict.
```

- Example 1: Use Dumpling + TiDB Lightning to import sharded databases and tables into TiDB in parallel
- Example 2: Import single tables in parallel

### Restrictions

> TiDB Lightning exclusively uses some resources when running. If you need to deploy multiple TiDB Lightning instances on a single machine (which is not recommended for production environments), or on a disk shared by multiple machines, you need to be aware of the following usage restrictions.

Reviewer: ```suggestion
TiDB Lightning exclusively uses some resources when it is running. If you need to deploy multiple TiDB Lightning instances on a single machine (which is not recommended for production environments), or on a disk shared by multiple machines, be aware of the following usage restrictions.
```
